tinygrad/runtime/graph/hcq.py

```diff
@@ -54,7 +54,11 @@ def __init__(self, jit_cache: List[ExecItem], input_rawbuffers: List[Buffer], va
 
     for j,ji in enumerate(jit_cache):
       enqueue_dev = ji.prg.dev if (is_exec_prg:=isinstance(ji.prg, CompiledRunner)) else Device[ji.bufs[1].device] #type:ignore
-      enqueue_queue = self.comp_queues[enqueue_dev] if is_exec_prg else self.copy_queues.setdefault(enqueue_dev, enqueue_dev.hw_copy_queue_t())
+      assert isinstance(enqueue_dev, HCQCompiled)
+      if is_exec_prg: enqueue_queue = self.comp_queues[enqueue_dev]
+      else:
+        assert issubclass(enqueue_dev.hw_compute_queue_t, HWQueue) and enqueue_dev.hw_copy_queue_t is not None
+        enqueue_queue = self.copy_queues.setdefault(enqueue_dev, enqueue_dev.hw_copy_queue_t())
```

Comment:
This is unreadable and needs refactoring

Reply:
Going to open up a new PR with the other changes and convert this one to draft.

Reply:
Let me know if it is readable enough now.
Unfortunately, mypy doesn't have the path sensitivity for something with as few lines as possible like:
```python
if not is_exec_prg: assert (enqueue_dev.hw_copy_queue_t is not None)
enqueue_queue = self.comp_queues[enqueue_dev] if is_exec_prg else self.copy_queues.setdefault(enqueue_dev, enqueue_dev.hw_copy_queue_t())
```